Rubberduck.VBEEditor/Events/VBENativeServices.cs

```diff
@@ -57,136 +45,94 @@ public static void UnhookEvents()
             lock (ThreadLock)
             {
                 User32.UnhookWinEvent(_eventHandle);
-                foreach (var info in TrackedWindows.Values)
-                {
-                    info.Subclass.FocusChange -= FocusDispatcher;
-                    info.Subclass.Dispose();
-                }
+                Subclasses.Dispose();
                 VBEEvents.Terminate();
+                _vbe.Dispose();
                 _vbe = null;
             }
         }
 
+        private static void Attach(IntPtr hwnd)
+        {
+            var subclass = Subclasses.Subclass(hwnd);
+
+            if (subclass == null)
+            {
+                return;
+            }
+
+            if (subclass is IFocusProvider focusSource)
+            {
+                focusSource.FocusChange += FocusDispatcher;
+            }
+
+            if (subclass is IWindowEventProvider keyboardListener)
+            {
+                keyboardListener.KeyDown += KeyDownDispatcher;
+            }
+        }
+
         public static void VbeEventCallback(IntPtr hWinEventHook, uint eventType, IntPtr hwnd, int idObject, int idChild,
             uint dwEventThread, uint dwmsEventTime)
         {
             if (hwnd == IntPtr.Zero) { return; }
-            //This is an output window firehose, leave this here, but comment it out when done.
-            //if (idObject != (int)ObjId.Cursor) { Debug.WriteLine("Hwnd: {0:X4} - EventType {1:X4}, idObject {2}, idChild {3}", (int)hwnd, eventType, idObject, idChild); }
 
+#if THIRSTY_DUCK
```

Comment:
lol for that compilation flag name. Do we need to explicitly set it? Should we maybe also check for `DEBUG` having been set?

Reply:
I must not have been thirsty enough to check that on a release build... Fixed.